chainer/dataset/dataset_mixin.py

```diff
@@ -15,18 +19,49 @@ class DatasetMixin(object):
     def __getitem__(self, index):
         """Returns an example or a sequence of examples.
 
-        It implements the standard Python indexing. It uses the
-        :meth:`get_example` method by default, but it may be overridden by the
-        implementation to, for example, improve the slicing performance.
+        It implements the standard Python indexing and one-dimensional integer
+        array indexing. It uses the :meth:`get_example` method by default, but
+        it may be overridden by the implementation to, for example, improve the
+        slicing performance.
+
+        Args:
+            index (int, slice, list or numpy.ndarray): An index of an example
+                or indexes of examples.
+
+        Returns:
+            If index is int, returns an example created by `get_example`.
+            If index is either slice or one-dimensional list or numpy.ndarray,
+            returns a list of examples created by `get_example`.
+
+        .. admonition:: Example
+           >>> import numpy
```

Comment:
Please insert one blank line right before this line (L:37), otherwise this Example part doesn't show up in the built docs.